RIOT-OS/RIOT
# boards/common/remote: fix JLink flashing and debug

### Contribution description

This PR fixes the flashing command using JLink, which is not currently working.
It makes use of the jlink.inc.mk Makefile which in turn configure and uses dist/tools/jlink/jlink.sh.
Flashing works correctly and is faster than cc2538-bsl, however it needs an external JLink adapter and to solder the header on the remote.

Additionally, I modified the reset command to use JLink by default, since the cc2538-bsl script is not very handy to perform `make test`. With JLink that command works correctly and thus we can use it for CI HIL tests.

### Testing procedure


I've tested using a remote-revb.

`PROGRAMMER=jlink BOARD=remote-revb make -C tests/shell/ flash` should use JLink as the flasher.

`make reset` should now by default use JLink, therefore, in order to continue using cc2538-bsl one needs to override the "reseter" as following: `RESETER=cc2538-bsl`.

### Issues/PRs references


#9428 added reset and this PR overrides the default reseter.
